[PyTorch] Add unit test for c10::Synchronized<T>

dhruvbird opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #74062
* #74061
* #74060
* #73725
* #73724
* #73723

Since this class was added, it's missing a unit test. @dbort first noticed it, so adding one in this commit.

Differential Revision: [D34800969](https://our.internmc.facebook.com/intern/diff/D34800969/)

dhruvbird committed:
[PyTorch] Add unit test for c10::Synchronized<T>

Since this class was added, it's missing a unit test. @dbort first noticed it, so adding one in this commit.

Differential Revision: [D34800969](https://our.internmc.facebook.com/intern/diff/D34800969/)

[ghstack-poisoned]

dbort commented on c10/test/util/Synchronized_test.cpp:
```diff
@@ -0,0 +1,50 @@
+#include "c10/util/Synchronized.h"
+#include <c10/util/Synchronized.h>
```
Duplicate includes

dhruvbird replied:
Thanks VSCode!

dbort reviewed (commented):
Thank you so much for adding this! Looks great.

Please take a look at the clang-tidy and clang-format warnings/errors on internal CI.

dhruvbird committed:
Update on "[PyTorch] Add unit test for c10::Synchronized<T>"


Since this class was added, it's missing a unit test. dbort first noticed it, so adding one in this commit.

Differential Revision: [D34800969](https://our.internmc.facebook.com/intern/diff/D34800969/)

[ghstack-poisoned]

dhruvbird committed:
Update on "[PyTorch] Add unit test for c10::Synchronized<T>"


Since this class was added, it's missing a unit test. dbort first noticed it, so adding one in this commit.

Differential Revision: [D34800969](https://our.internmc.facebook.com/intern/diff/D34800969/)

[ghstack-poisoned]

malfet commented:
Why this change was landed when it clearly broke Windows builds and clang-format?
